f5_openstack_agent/lbaasv2/drivers/bigip/test/test_exceptions.py

```diff
@@ -0,0 +1,533 @@
+#!/usr/bin/env python
+# Copyright 2017 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import inspect
+import logging
+import pdb
+import pytest
+import sys
+import traceback
+
+import f5_openstack_agent.lbaasv2.drivers.bigip.exceptions as exceptions
+
+from collections import namedtuple
+from mock import Mock
+from mock import mock_open
+from mock import patch
+
+
+class Test_F5MissingDependencies(object):
+    def setup_traceback(self):
+        traceback.print_exc = Mock()
+
+    def setup_iterations(self):
+        Options = \
+            namedtuple('Options',
+                       'errno, message, details, exception, frame')
+        set_frame = inspect.getframeinfo(inspect.currentframe())
+        items = [Options(None, None, [], None, None),
+                 Options(22, 'missing', [31415926], ImportError('foodogzoo'),
+                         set_frame),
+                 Options(22, 'missing', [31415926], None, set_frame)]
+        logging.FileHandler = Mock()
+        logger = Mock()
+        logging.getLogger = Mock(return_value=logger)
+        logger.exception = Mock()
+        logger.addHandler = Mock()
+        sys.argv.append('f5-openstack-agent.ini')
+        return items, set_frame
+
+    def test_iterations(self):
+        test_items, set_frame = self.setup_iterations()
+        self.setup_traceback()
+        for tset in test_items:
+            message = tset.message
+            errno = tset.errno
+            details = tset.details
+            exc = tset.exception
+            fr = tset.frame
+            read_lines = 'debug = true\n'
+            m_open = mock_open(read_data=read_lines)
+            open_name = \
+                'f5_openstack_agent.lbaasv2.drivers.bigip.exceptions.open'
```

Comment:
Let's make `open_name` a class variable and reference it as `self.open_name` so we don't have to assign it more than once.